hbase-common/src/main/java/org/apache/hadoop/hbase/io/crypto/Encryption.java

```diff
@@ -578,4 +627,45 @@ public static void incrementIv(byte[] iv, int v) {
     } while (v > 0);
   }
 
+  /**
+   * Return the hash of the concatenation of the supplied arguments, using the 
+   * hash algorithm provided.
+   */
+  public static byte[] hashWithAlg(String algorithm, byte[]... args) {
+    try {
+      MessageDigest md = MessageDigest.getInstance(algorithm);
+      for (byte[] arg: args) {
+        md.update(arg);
+      }
+      return md.digest();
+    } catch (NoSuchAlgorithmException e) {
+      throw new RuntimeException("unable to use hash algorithm: " + algorithm, e);
+    }
+  }
+
+  private static byte[] hashWithAlg(String algorithm, String... args) {
```

Comment:
Refactor to re-use hashWithAlg() with byte[] or String?

Reply:
Originally I decided against it for performance reasons, I wanted to avoid the creation of a new long string or byte array. I wanted to have the same memory footprint as the original implementation had, as this was a function on the public API. 

However, I don't think anyone is using this function on any critical path. And your suggestion makes the code cleaner. I'll change this. I also found the `Bytes.toByteArrays` function so I can fully eliminate this private function.